search/search_quality/assessment_tool/main_model.cpp

```diff
@@ -64,24 +72,53 @@ void MainModel::OnSampleSelected(int index)
     params.m_suggestsEnabled = false;
     params.SetPosition(latLon.lat, latLon.lon);
 
-    auto const timestamp = ++m_queryTimestamp;
+    auto const timestamp = m_queryTimestamp;
     m_numShownResults = 0;
 
-    params.m_onResults = [this, timestamp](search::Results const & results) {
-      GetPlatform().RunOnGuiThread([this, timestamp, results]() { OnResults(timestamp, results); });
+    params.m_onResults = [this, sample, timestamp](search::Results const & results) {
```

Comment:
The style you proposed is quite arguable. When you pass callbacks on other threads it's quite dangerous to capture everything. Much safer to explicitly declare all what you need.